igvc/src/motor_controller/main.cpp

```diff
@@ -28,66 +35,164 @@ int main(int argc, char** argv)
-
   ros::Publisher enc_pub = nh.advertise<igvc_msgs::velocity_pair>("/encoders", 1000);
 
+  ros::Publisher enabled_pub = nh.advertise<std_msgs::Bool>("/robot_enabled", 1);
+
+  ros::Publisher battery_pub = nh.advertise<std_msgs::UInt8>("/battery", 1);
+
   std::string device_path;
-  nhp.param(std::string("device"), device_path, std::string("/dev/igvc_motor_arduino"));
+  nhp.param(std::string("device"), device_path, std::string("/dev/igvc_motor_board"));
 
   int baud_rate;
   nhp.param(std::string("baud_rate"), baud_rate, 9600);
 
+  int messages_to_read;
+  nhp.param(std::string("messages_to_read"), messages_to_read, 3);
+
+  nhp.param("precision", precision, 1);
+
+  double p_l, p_r, d_l, d_r, i_l, i_r;
+  nhp.param("p_l", p_l, 3.0);
+  nhp.param("p_r", p_r, 3.0);
+  nhp.param("d_l", d_l, 0.0);
+  nhp.param("d_r", d_r, 0.0);
+  nhp.param("i_r", i_r, 0.0);
+  nhp.param("i_l", i_l, 0.0);
+
+  ROS_INFO_STREAM("Opening port");
   SerialPort port(device_path, baud_rate);
+  ROS_INFO_STREAM("Opened port");
 
   if (!port.isOpen())
   {
-    ROS_ERROR_STREAM("Motor Controller serial port failed to open.");
+    ROS_ERROR_STREAM("Motor Board serial port failed to open.");
     return -1;
   }
   port.flush();
-  ROS_INFO_STREAM("Motor Controller ready.");
+  ROS_INFO_STREAM("Motor Board ready.");
 
   ros::Rate rate(10);
-  while (ros::ok() && port.isOpen())
-  {
-    ros::spinOnce();
 
-    std::string msg = "$" + std::to_string(enabled ? current_motor_command.left_velocity : 0.0) + "," +
-                      std::to_string(enabled ? current_motor_command.right_velocity : 0.0) + "\n";
+  std::string p_values = "#P" + toBoundedString(p_l) + "," + toBoundedString(p_r) + "\n";
+  std::string d_values = "#D" + toBoundedString(d_l) + "," + toBoundedString(d_r) + "\n";
+  std::string i_values = "#I" + toBoundedString(i_l) + "," + toBoundedString(i_r) + "\n";
 
-    if (fabs(current_motor_command.left_velocity) > 0 || fabs(current_motor_command.right_velocity) > 0)
-      ROS_ERROR_STREAM("msg to motor arduino $lv,rv\n" << msg);
+  bool valid_values_p = false;
+  bool valid_values_d = false;
+  bool valid_values_i = false;
+
+  while(ros::ok() && (!valid_values_p || !valid_values_d || !valid_values_i)) {
+    ros::spinOnce();
+    ROS_INFO_STREAM("sending PID values to board");
+    if(!valid_values_p) {
+      port.write(p_values);
+    }
+    if(!valid_values_d) {
+      port.write(d_values);
+    }
+    if(!valid_values_i) {
+      port.write(i_values);
+    }
+    for(int i = 0; i < 3; i++) {
+      std::string ret = port.readln();
+      if(!ret.empty() && ret.at(0) == '#' && ret.at(1) != 'E') {
+        size_t p_loc = ret.find('P');
+        size_t d_loc = ret.find('D');
+        size_t i_loc = ret.find('I');
+        size_t end = ret.find('\n');
+        if(p_loc != std::string::npos) {
+          std::vector<std::string> vals = split(ret.substr(p_loc + 1, end), ',');
+          ROS_INFO_STREAM("p ret = " << ret);
```

Comment:
This seems like a debug print statement.  I think we could either print nothing if everything is OK with the PID constants returned by the motor board or if there's an issue, print the expected vs. actual values sent back from the motor board